### DISCUSSION TITLE
32mhz clock not working

### DISCUSSION THREAD

**Discussion by Carlosfandango22** [dbuezas/lgt8fx#160] [General]
Hello, I have 10 of these LGT8F328P LQFP32 MiniEVB boards which I have just received from AliExpress. 
https://www.aliexpress.com/item/1005001274006314.html?spm=a2g0s.9042311.0.0.27424c4dm35rn8

I'm using these with the Arduino IDE 1.8.15. They seem to work fine at 16MHz. As soon as I select any variation of 32MHz by changing the clock source, clock speed or both, the code will upload but the boards become unresponsive. I feel like I must be doing something wrong? Is anyone able to tell me if I'm suppose to add a library, or include anything within my code to get it working at 32MHz?

For a test I'm simply serial printing "hello" with a delay of 1000ms.

Thanks,
Carl

**Comment by dbuezas** [comment]
That's odd... there's no lib to include.

 You have the green ones with the micro usb, right?
And you select the internal clock, right?

Did you install the core through writing the json in the arduino configuration?

**Reply by Carlosfandango22** [reply]
I'm convinced I am doing something wrong. I will see If I can find a known working board.

That's right, they the LQFP32 MiniEB boards with micro usb.
I have tried all the variations of clock speed/internal, eternal clock...

I installed it exactly as Githib says, through adding the json in the Arduino config. 

It is so odd.

**Reply by Carlosfandango22** [reply]
I have tried it with a mini evb and that seems to work at 32MHz... Sadly I can't find any of my older stock of the nano type LQFP32 to see if it is a problem with the boards...I am starting to suspect it probably is.

**Reply by LaZsolt** [reply]
Is there any differences between the non working and the working board? (on 32 MHz)
(Different components or different writings on back side.)

**Reply by Carlosfandango22** [reply]
No there is no difference at all... exactly the same to the eye... I haven't tested the resistor values yet though. The one which isn't working is from a recent batch of boards I bought this year. The one which is working is a couple of years old now.

**Reply by LaZsolt** [reply]
My suspicion is that your MCU is D variant. This variant ( LGT8F328D ) unable work on 32 MHz.
I made a qiuck test which tells your MCU type. Select internal clock speed to 16 MHz or lower, before upload.
```
//Quick test to decide the type of the LGT8F328 MCU.
void setup() {
  Serial.begin(38400);

#define TESTPORTEP _SFR_IO8(0x0E)
#define TESTPORTED _SFR_MEM8(0xA9)

uint8_t x;

  x = 0xF0;
  TESTPORTED = x;
  Serial.print(F("1"));
  Serial.print(F(". Port E work like LGT8F328 "));
  if (TESTPORTEP == x) Serial.println(F("P")); else Serial.println(F("D"));
  x = 0xA5;
  TESTPORTED = x;
  Serial.print(F("2"));
  Serial.print(F(". Port E work like LGT8F328 "));
  if (TESTPORTEP == x) Serial.println(F("P")); else Serial.println(F("D"));

  Serial.println();
#ifndef VCAL
  #define VCAL  _SFR_MEM8(0xC8)
#endif
#ifndef VCAL1
  #define VCAL1 _SFR_MEM8(0xCD)
#endif
#ifndef VCAL2
  #define VCAL2 _SFR_MEM8(0xCE)
#endif

  VCAL = VCAL2;
  Serial.print(F("Reference voltage calibration constant test.\nIt's working like LGT8F328 "));
  if (VCAL1 == VCAL2) {
    Serial.println(F("D"));
  }
  VCAL = VCAL1;
  if (VCAL != VCAL2) {
    Serial.println(F("P"));
  }
  Serial.println();
}

void loop() {}
```

**Reply by dbuezas** [reply]
Wow!

**Comment by dbuezas** [comment]
If you have one lgt328p working or a uno laying around, you could try the ISP method.
Maybe they have a bootloader problem

https://github.com/rickygai/arduino/blob/main/LGT8F328P.pdf

**Reply by Carlosfandango22** [reply]
I will try this and update if it solves the problem or not. Thank you very much.
